builder/lxc/step_lxc_create.go

```
@@ -14,54 +16,90 @@ import (
 	ui := state.Get("ui").(packer.Ui)
+	errorHandler := func(err error) {
+		state.Put("error", err)
+		ui.Error(err.Error())
+	}
 
-	name := config.ContainerName
-
-	// TODO: read from env
-	lxc_dir := "/var/lib/lxc"
-	rootfs := filepath.Join(lxc_dir, name, "rootfs")
-
-	if config.PackerForce || config.CleanupFirst {
+	if config.PackerForce {
 		s.destroy(config.ContainerName, ui)
 	}
 
-	commands := make([][]string, 3)
-	if config.CloneSource == "" {
-		commands[0] = append(config.EnvVars, []string{"lxc-create", "-n", name, "-t", config.Name, "--"}...)
-		commands[0] = append(commands[0], config.Parameters...)
+	var rootfs string
+	var err error
+	if config.LxcTemplate.Name != "" {
+		ui.Say("Creating container from template...")
+		rootfs, err = s.createFromTemplate(config.ContainerName, config.LxcTemplate)
 	} else {
-		commands[0] = []string{"lxc-clone", "--orig", config.CloneSource, "--new", name}
+		ui.Say(fmt.Sprintf("Creating container from archive: %s", config.RootFs.Archive))
+		rootfs, err = s.createFromRootFs(config.ContainerName, config.RootFs)
 	}
-	// prevent tmp from being cleaned on boot, we put provisioning scripts there
-	// todo: wait for init to finish before moving on to provisioning instead of this
-	commands[1] = []string{"touch", filepath.Join(rootfs, "tmp", ".tmpfs")}
-	commands[2] = []string{"lxc-start", "-d", "--name", name}
-
-	ui.Say("Creating container...")
-	for _, command := range commands {
-		log.Printf("Executing sudo command: %#v", command)
-		err := s.SudoCommand(command...)
-		if err != nil {
-			err := fmt.Errorf("Error creating container: %s", err)
-			state.Put("error", err)
-			ui.Error(err.Error())
-			return multistep.ActionHalt
-		}
+	if err != nil {
+		errorHandler(err)
+		return multistep.ActionHalt
+	}
+	ui.Say("Starting container...")
+	if err = s.SudoCommand("lxc-start", "-d", "-n", config.ContainerName); err != nil {
+		errorHandler(fmt.Errorf("Error starting container: %s", err))
```

I think this missed getting updated as well

---

I'm not sure what's up, but `errorHandler` is defined as an anon function. This branch compiles fine for me....

---

What the. Yeah, you define `errorHandler := func(err error) { ...` but then pass two arguments here. You could probably just break out the function since it's only called twice for two lines but either way why does this line compile? :S

---

`errorHandler` is called with one argument: `fmt.Errorf("Error starting container: %s", err)`. `fmt.Error` is the one being called with 2 args.

---

Oh, phf, my bad. Should we not still be passing the error through like we do for the other call? I don't want to keep holding you up for it though, it's not a big difference in the large scheme of things.

---

Hmm, I don't think we want to return the error in this case. The `Run` function needs to return a `multistep.StepAction`. That's obligatory.

So wrapping the error just provides more context on what the plugin was doing on top of the error relating to the command itself.